### PR TITLE
[android][expo-battery] Add method to check if battery optimization is enabled

### PR DESCRIPTION
**Pull request by FelipeACP**
# Why

Many manufacturers restrict background tasks in favor of longer battery life. They make battery optimization default for every app. Therefore, it's useful to know whether it is enabled for your app so that you can handle your logic accordingly. 

Example issues where battery optimization disturbed the normal behavior:  #7485 #10700 #8830

# How

PowerManager has one public method called `isIgnoringBatteryOptimizations`  which "returns whether the given application package name is on the device's power allowlist". So this PR basically adds a wrapper for the given method with the opposite logic.

# Test Plan

Run the test app, on Battery Screen, check if `batteryOptimizationState` matches the battery optimization setting.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-battery/android/src/main/java/expo/modules/battery/BatteryModule.java [hunk 1]
```diff
@@ -142,6 +143,19 @@ public void isLowPowerModeEnabledAsync(Promise promise) {
     promise.resolve(isLowPowerModeEnabled());
   }
 
+  @ExpoMethod
+  public void isBatteryOptimizationEnabledAsync(Promise promise) {
+    if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```
Comment (tsapeta) [inline packages/expo-battery/android/src/main/java/expo/modules/battery/BatteryModule.java:148]:
```suggestion
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

File: packages/expo-battery/src/Battery.ts [hunk 2]
```diff
@@ -80,6 +80,24 @@ export async function isLowPowerModeEnabledAsync(): Promise<boolean> {
...
+  * ```ts
+  * await Battery.isBatteryOptimizationEnabledAsync();
+  * // true
+  * ```
+  */
+  export async function isBatteryOptimizationEnabledAsync(): Promise<boolean> {
+    if (!ExpoBattery.isBatteryOptimizationEnabledAsync) {
+        return false;
+    }
+    return await ExpoBattery.isBatteryOptimizationEnabledAsync();
```
Comment (tsapeta) [inline packages/expo-battery/src/Battery.ts:98]:
```suggestion
  if (!ExpoBattery.isBatteryOptimizationEnabledAsync) {
    return false;
  }
  return await ExpoBattery.isBatteryOptimizationEnabledAsync();
```

**Review by tsapeta — CHANGES_REQUESTED** [review]
Except some formatting stuff, it looks good! Thanks for your contribution 🙇‍♂️ 
Would you mind implementing this on iOS too? Looks like we can use [NSProcessInfo.processInfo.lowPowerModeEnabled](https://developer.apple.com/documentation/foundation/nsprocessinfo/1617047-lowpowermodeenabled?language=objc) for this 🤔

**Comment by FelipeACP** [comment]
> Except some formatting stuff, it looks good! Thanks for your contribution 🙇‍♂️
> Would you mind implementing this on iOS too? Looks like we can use [NSProcessInfo.processInfo.lowPowerModeEnabled](https://developer.apple.com/documentation/foundation/nsprocessinfo/1617047-lowpowermodeenabled?language=objc) for this 🤔
No, I wouldn't mind! The reason I haven't done so already is because AFIK iOS doesn't discriminate against any app battery-wise, that is, the user cannot enable or ignore battery optimizations for specific apps. 
On the other hand, on Android, you have a list of app that ignore battery optimizations, so this method is essential to know beforehand if the OS might restrict your background activities when the device go into doze mode.

**Comment by tsapeta** [comment]
Ok, gotcha! I misunderstood the intention at the beginning, but it makes sense after your explanation 😉 I think it's fine as if only `false` is returned on other platforms. Thanks!

**Review by tsapeta — APPROVED** [review]

**Pull request merged**